common/test/java/com/couchbase/lite/internal/core/C4DatabaseTest.kt

```diff
@@ -0,0 +1,728 @@
+    }
+
+    @Test
+    fun testDatabaseCopyToNonexistentDirectoryFails() {
+        try {
+            C4Database.copyDb(
+                c4Database.dbPath!!,
+                File(File(getScratchDirectoryPath(getUniqueName("a")), "aa"), "aaa").path,
+                getUniqueName("c4_copy_test_db"),
+                flags,
+                C4Constants.EncryptionAlgorithm.NONE,
+                null)
+            fail("Copy to non-existent directory should fail")
+        } catch (ex: LiteCoreException) {
+            assertEquals(C4Constants.ErrorDomain.LITE_CORE.toLong(), ex.domain.toLong())
+            assertEquals(C4Constants.LiteCoreError.NOT_FOUND.toLong(), ex.code.toLong())
+        }
+    }
+
+    @Test
+    fun testDatabaseCopyFromNonexistentDbFails() {
+        try {
+            C4Database.copyDb(
+                File(c4Database.dbPath?.let { File(it).parentFile }, "x" + C4Database.DB_EXTENSION).path,
+                getScratchDirectoryPath(getUniqueName("c4_test_2")),
+                getUniqueName("c4_copy_test_db"),
+                flags,
+                C4Constants.EncryptionAlgorithm.NONE,
+                null)
+            fail("Copy from non-existent database should fail")
+        } catch (ex: LiteCoreException) {
+            assertEquals(C4Constants.ErrorDomain.LITE_CORE, ex.domain)
+            assertEquals(C4Constants.LiteCoreError.NOT_FOUND, ex.code)
+        }
+    }
+
+    @Test
+    @Throws(LiteCoreException::class)
+    fun testDatabaseCopyToExistingDBFails() {
+        createRev("doc001", REV_ID_1, fleeceBody)
+        createRev("doc002", REV_ID_1, fleeceBody)
+        val srcDbPath = c4Database.dbPath
+        val dstParentDirPath = getScratchDirectoryPath(getUniqueName("c4_test_2"))
+        var targetDb = C4Database.getDatabase(
+            dstParentDirPath,
+            dbName,
+            flags,
+            C4Constants.EncryptionAlgorithm.NONE,
+            null)
+        createRev(targetDb, "doc001", REV_ID_1, fleeceBody)
+        assertEquals(1, targetDb.documentCount)
+        targetDb.close()
+        try {
+            C4Database.copyDb(
+                srcDbPath!!,
+                dstParentDirPath,
+                dbName,
+                flags,
+                C4Constants.EncryptionAlgorithm.NONE,
+                null)
+        } catch (ex: LiteCoreException) {
+            assertEquals(C4Constants.ErrorDomain.POSIX, ex.domain)
+            assertEquals(POSIX_EEXIST, ex.code)
+        }
+        targetDb = C4Database.getDatabase(
+            dstParentDirPath,
+            dbName,
+            flags,
+            C4Constants.EncryptionAlgorithm.NONE,
+            null)
+        assertEquals(1, targetDb.documentCount)
+        targetDb.close()
+    }
+
+
+    // - Scopes and Collections
+
+    //  Test create 1 scope and 1 collection
+    @Test
+    fun testCreateScopeAndCollection() {
+        c4Database.addCollection("test_scope", "test_coll")
```

Comment:
can you check to be sure the size of scopeNames is 0, first?

Reply:
should scopeNames size be 1 first, since there'll always be _default scope, unless user deletes it